File: Week_03/G20190343020044/leetcode-169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) <= 2:
            return nums[0]
        x = len(nums)/2
        d = {}
        for i in nums:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
                if d[i] >= x:
                    return i

File: Week_03/G20190343020074/LeetCode_169.java
class Solution {
    public int majorityElement(int[] nums) {
        int countFlag=0;
        int result = 0;
        for(int i=0;i<nums.length;i++)
        {
            if(countFlag==0){
                result=nums[i];
            }
            countFlag+=(result==nums[i])?1:-1;
        }
        return result;
    }
}
